varlink/connection.go

```diff
@@ -76,6 +76,13 @@ func (e *Error) Error() string {
 	return e.Name
 }
 
+// ReadWriterContext describes the capabilities of the
+// underlying varlink connection.
+type ReadWriterContext interface {
+	WriterContext
+	ReadBytes(ctx context.Context, delim byte) ([]byte, error)
```

Comment:
To consume the data podman provides, I'm using code very similar to https://github.com/containers/libpod/blob/b713e5371fa8767067bd2c329a3744ea593b2ade/pkg/varlinkapi/virtwriter/virtwriter.go .  I'd thus need the `varlink.ReadWriterContext` to implement the `io.Reader` interface. Is that possible?

Reply:
I'd rather not expose this since the context is there for a reason. I could expose a `ReadContext(ctx context.Context, []byte) (int, error)` and you can wrap that:
```go
type ReaderContext interface {
    ReadContext(context.Context, []byte) (int, error)
}

type readerCtx struct {
    ReaderContext
    ctx context.Context
}

func (r readerCtx) Read(in []byte) (int, error) {
    return r.ReaderContext.ReadContext(r.ctx, in)
}
```

Use like so:
```go
r := readerCtx{
    ReaderContext: conn,
    ctx: ctx,
}
// r implements io.Reader, but still supports cancellation
```

Reply:
Done

Reply:
Thanks @johanbrandhorst , it should be indeed good enough for my use case. Let me experiment with my code, I'll report ASAP.